src/main/java/org/ozwillo/dcexporter/service/CkanClientService.java

```diff
@@ -36,6 +36,20 @@
             return Optional.empty();
         }
     }
+    
+    public Optional<List<CkanDataset>> getCompleteDatasets(String ckanUrl, int limit) {
+        if(!isValid(ckanUrl, "CKAN URL")) return Optional.empty();
+
+        CkanAPI ckanAPI = getCkanAPI(ckanUrl);
+        Call<DatasetListResponse> call = ckanAPI.getCompleteDatasets(limit);
+
+        try {
+            return Optional.ofNullable(call.execute().body().result);
+        } catch (IOException e) {
+            LOGGER.error("Error while trying to fetch datasets from CKAN: {}", e);
+            return Optional.empty();
+        }
+    }
 
```

Comment:
New method to call `current_package_list_with_resources` CKAN action.
This API function returns the list of packages (Dataset) with complete information.
Pb : by default, it returns 10 results. This can be bypassed with limit parameter